api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -265,7 +273,14 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
       Workspace workspace) {
     org.pmiops.workbench.db.model.Workspace dbWorkspace = workspaceService.getRequired(
```

Comment:
Probably most of this logic should move into a new updateWorkspace() method on WorkspaceService.

This will also allow you to reuse logic for ETag checking, etc.